images/devtools-terraform-v1beta1/context/Dockerfile

```diff
@@ -50,25 +52,27 @@ COPY --from=tfsec  /usr/bin/tfsec /usr/bin/tfsec
 COPY --from=golang /usr/local/bin/goenv.sh /usr/local/bin/goenv.sh
 COPY --from=docker-lock /prod/docker-lock /usr/local/bin/docker-lock
 
-COPY terraform-switcher_${tfswitcher_version}_checksums.txt /var/tmp/terraform-switcher_${tfswitcher_version}_checksums.txt
+COPY terraform_switcher_checksums.txt terraform_switcher_checksums.txt
 
 # Setting TF_PLUGIN_CACHE_DIR makes things a lot faster but this somehow
 # causes weird permission issues.
 # ENV TF_PLUGIN_CACHE_DIR=/root/.cache/terraform.d/plugin-cache
 
-ARG terraform_versions="0.15.5 1.0.11 1.1.7"
-
 RUN \
     mkdir -vp ~/.cache ~/.local/bin ~/bin && \
     true
 
-RUN \
-    cd /var/tmp && \
-    . /usr/local/bin/goenv.sh && \
-    curl -LOJ "https://github.com/warrensbox/terraform-switcher/releases/download/${tfswitcher_version}/terraform-switcher_${tfswitcher_version}_${GOOS}_${GOARCH}.tar.gz" && \
-    sed -i "/_${GOOS}_${GOARCH}.tar.gz/!d" "terraform-switcher_${tfswitcher_version}_checksums.txt" && \
-    sha256sum -c "terraform-switcher_${tfswitcher_version}_checksums.txt" && \
-    tar -xvf "terraform-switcher_${tfswitcher_version}_${GOOS}_${GOARCH}.tar.gz" -C /usr/local/bin tfswitch && \
+RUN --mount=type=secret,id=github_token \
+    if [[ -e "/run/secrets/github_token" ]]; then GITHUB_AUTH="-u \"$(cat /run/secrets/github_token):x-oauth-basic\""; else GITHUB_AUTH=""; fi && \
+    if [[ $tfswitcher_version == "" ]]; then TERRAFORM_SWITCHER_TAG="latest"; else TERRAFORM_SWITCHER_TAG="tags/tfswitcher_version"; fi && \
```

Comment:
This should not use latest at all, as per guideline:

https://playbook.internal.coop/guidelines/security/supply_chain.html?h=lock#version-locking

> Lock the versions of your dependencies. When locking the version of a dependency a checksum for the exact version of the dependency is store in your source code repository and used to reinstall the exact same version.

This basically adds a potential for this not be the case, which is not desirable IMO.

Reply:
> This should not use latest at all, as per guideline:
> 
> https://playbook.internal.coop/guidelines/security/supply_chain.html?h=lock#version-locking
> 
> > Lock the versions of your dependencies. When locking the version of a dependency a checksum for the exact version of the dependency is store in your source code repository and used to reinstall the exact same version.
> 
> This basically adds a potential for this not be the case, which is not desirable IMO.

Fair, will pin the version.

Reply:
But in this case, the dockerfile should only operate on a pinned version so we don't have a potential failure mode for downloading an arbitrary version.